src/Autocomplete.php

```diff
@@ -137,12 +137,14 @@ public function registerEventHandlers()
      */
     public function generateAutocompleteClasses()
     {
-        $autocompleteGenerators = $this->getAllAutocompleteGenerators();
-        foreach ($autocompleteGenerators as $generatorClass) {
-            /* @var Generator $generatorClass */
-            $generatorClass::generate();
+        if (Craft::$app->getIsInstalled()) {
```

Comment:
It looks like at this point, no primary site exists. Does this necessarily mean the plugin is not installed, or couldn’t the error also appear in an installed setup with no primary sites?

Reply:
Well the thing is, Craft Autocomplete is a self-bootstrapping module, so as long as it's required, it's installed.

Reply:
Sorry I meant Craft (not a plugin). What I’m getting at is, wouldn’t it be more correct to check whether a primary site exists rather than whether Craft is installed?

Reply:
Doesn't the installation process create the default site? Although I guess you're right, someone _could_ potentially delete the one and only site somehow.

Reply:
I'm gonna merge it anyway, because I think this is probably a good safety check to have in regardless. If we need to do more for further edge cases down the road, we can.